### DISCUSSION TITLE
IPv6 Network setup suggestion

### DISCUSSION THREAD

**Discussion by Zebradil** [k3s-io/k3s#9045] [Q&A] [answered]
I want to install k3s in the following environment (it's a home lab, basically):
- 2 nodes (server and agent) in LAN
- dynamic global IPv6 addresses that are used for reaching out to the cluster
- low traffic

So far I have figured out that k3s doesn't work with dynamic IPv6 addresses (at least, in the default dual-stack configuration).

I'm thinking about re-creating the cluster in IPv4-only mode and using the host network for traefik to make it respond on IPv6. I can bind traefik to a particular node, high availability is not required.

Does it make sense? Is there maybe some non-default component that can handle dynamic IPv6 addresses?
I'd appreciate some thoughts on this and input from experienced users.

**Comment by brandond** [comment, marked as answer]
Dynamic IPs are in general a bad idea when it comes to Kubernetes. The assumption is that nodes will not change IPs while they are joined to the cluster.

Using Traefik to handle IPv6 requests from external clients using an ipv4 cluster network is probably a good compromise.

**Reply by Zebradil** [reply]
Thank you for the quick response. I'll try the host network option then.

<hr>
I dislike the dynamic IPv6 situation very much, tbh. But this is something I can't change. Days were spent on this issue in general (I had to write a dynamic DNS service to update my IPv6 addresses).
